graphql_compiler/compiler/ir_lowering_common.py

```diff
@@ -204,3 +205,55 @@ def extract_folds_from_ir_blocks(ir_blocks):
                 remaining_ir_blocks.append(block)
 
     return folds, remaining_ir_blocks
+
+
+def extract_location_to_optional_from_ir_blocks(ir_blocks):
```

Comment:
I don't get how the name of this function corresponds to the docstring. I don't see anything related to `@optional` in there